rfc/openfed__setDescription.md

```diff
@@ -0,0 +1,307 @@
+_e.g._, even when following "normal" WunderGraph Cosmo description rules.
+
+### content
+- Optional.
+- Does nothing if the `public` argument is set to `false`.
+- Allows the definition of a new description, _e.g._, one that is different to the actual description in the subgraph
+(which may be intended to be private/internal), or for a definition (such as an extension) that cannot define 
+descriptions.
+
+# Considerations/discussion points
+
+The following is a list of considerations and/or potential discussion points:
+
+## Single source of precedence/single public description
+The federated graph can only define a single description.
+This description is "normally" the longest description (the first of that length in a tie).
+If the `@openfed__setDescription` defines the argument `public` as `true` (the default value), that description will be 
+propagated to the federated graph.
+If the `@openfed__setDescription` defines the argument `content`, the value of the content argument will "override"
+what the "original" description.
+If more than one shared definition instance defines the `@openfed__setDescription` `public` argument as `true`, a
+composition error will be returned.
+Only one shared definition instance may declare a public description.
+
+## Parent level vs. child level
+If `@openfed__setDescription` is defined on a parent level, _e.g._, an `Object type`, should all its children (_i.e._,
+fields) "inherit" the public declaration?
+For example:
+```graphql
+# subgraph A
+
+  """
+  A.Query description.
+  """
+  type Query @openfed__setDescription {
+    """
+    A.Query.dummy description.
+    """
+    dummy: String! @shareable
+  }
+```
+
+```graphql
+# subgraph B
+
+  """
+  B.Query description, which is longer than A.
+  """
+  type Query {
+    """
+    B.Query.dummy description, which is longer than A.
+    """
+    dummy: String! @shareable
+  }
+```
+
+For these two subgraphs, should the result be:
+1. `@openfed__setDescription` is only considered for the explicit location on which it is defined:
+```graphql
+# federated graph
+
+  """
+  A.Query description.
+  """
+  type Query {
+    """
+    B.Query.dummy description, which is longer than A.
+    """
+    dummy: String! @shareable
+  }
+```
+
+2. `@openfed__setDescription` is "inherited" by its children:
+```graphql
+# federated graph
+
+  """
+  A.Query description.
+  """
+  type Query {
+    """
+    A.Query.dummy description.
+    """
+    dummy: String! @shareable
+  }
+```
+
+It's also possible a new argument, _e.g._, `inheritance: Boolean! = false`, could dictate whether the behaviour 
+defined on the parent is "inherited" by children, _e.g._, all `public: true` or all `public: false`.
+The child should also be able to override this behaviour by explicitly defining its own directive.
+I'm mostly of the opinion that descriptions should be explicit and granular, and inherited/coerced behaviour might be
```

Comment:
I believe explicit behavior is more predictable

The only question is about the situation when we set a description for a type from subgraph `A` but fields are still using descriptions from subgraph `C`

Reply:
It may also be cumbersome if, for example, an Object defines 50 Fields, but you just want that subgraph to set _all_ public descriptions , _i.e._, for the parent and its children.